packages/x-tree-view/src/useTreeItem2/useTreeItem2.ts

```diff
@@ -65,6 +70,11 @@ export const useTreeItem2 = <
         return;
       }
 
+      // Don't blur the root when switching to editing mode
+      if (status.editing || event.target.id !== idAttribute) {
```

Comment:
New idea:

Would a check like:

```tsx
contentRef.contains(document.activeElement)
```

work here?
It would be more versatile if people want to add other focusable DOM nodes inside their items.

If it does not work well, then ignore this comment, I think focus management of items with focusable DOM children is something we need to improve in the future.

Reply:
It could work I think, I'm trying it now 🛩️

Reply:
`contentRef` is null here, so it would not work

Reply:
I assume it's about knowing if the blur come from the focus being watched by the input, or just because the user clicked somewhere else.

Maybe you can use `document.activeElement.closest( ... );` with ... a CSS selector of the tree view root component. Either the className. Or a `data-...` attribute added on purpose

Reply:
> I'm trying to check if this is being called on the same item root or on an item that's on a higher leve

Would this utility function help you @noraleonte 

https://github.com/mui/mui-x/pull/12213/files#diff-496aa979fde270b056c9a377d09f8cdba0b3d8b42476fb8491964b6defc04e9c


```tsx
/**
 * Checks if the event target is in a descendant of this item.
 * This can prevent from firing some logic on the ancestors on the interacted item when the event handler is on the root.
 * @param {React.UIEvent} event The event to check
 * @param {HTMLElement | null} itemRoot The root of the item to check if the event target is in its descendants
 * @returns {boolean} Whether the event target is in a descendant of this item
 */
export const isEventTargetInDescendants = (event: React.UIEvent, itemRoot: HTMLElement | null) => {
  return itemRoot !== (event.target as HTMLElement).closest('*[role="treeitem"]');
};
```

I can extract it from the dnd PR

Reply:
That would be great!

Reply:
#13982

Reply:
If you need `to be able to pass `document.activeElement` instead, we can tweak the method to suit you needs